# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1289629564 18000
#      Sat Nov 13 01:26:04 2010 -0500
# Node ID 29b1f7d68665c68ad42a44d114d8a67ae7a3b053
# Parent  c463aed850b0778fbb5d71f5d4f6e0c982bf88d8
make filename property of figure objects writable

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2010-11-13  John W. Eaton  <jwe@octave.org>
+
+	* graphics.h.in (figure::properties::filename): Make writable.
+
 2010-11-12  John W. Eaton  <jwe@octave.org>
 
 	* Makefile.am (nodist_liboctinterp_la_SOURCES): Remove
 	$(OPT_HANDLERS) from  the  list.  Bug #31623.
 
 2010-11-11  John W. Eaton  <jwe@octave.org>
 
 	* data.cc: New tests for cat.  Enable 4 asserts that were disabled.
diff --git a/src/graphics.h.in b/src/graphics.h.in
--- a/src/graphics.h.in
+++ b/src/graphics.h.in
@@ -2806,17 +2806,17 @@ public:
       radio_property paperorientation , "{portrait}|landscape|rotated"
       color_property color , color_property (color_values (1, 1, 1), radio_values ("none"))
       array_property alphamap , Matrix (64, 1, 1)
       string_property currentcharacter r , ""
       handle_property currentobject r , graphics_handle ()
       array_property currentpoint r , Matrix (2, 1, 0)
       bool_property dockcontrols , "off"
       bool_property doublebuffer , "on"
-      string_property filename r , ""
+      string_property filename , ""
       bool_property integerhandle , "on"
       bool_property inverthardcopy , "off"
       callback_property keypressfcn , Matrix ()
       callback_property keyreleasefcn , Matrix ()
       radio_property menubar , "none|{figure}"
       double_property mincolormap , 64
       string_property name , ""
       bool_property numbertitle , "on"
